### PR TITLE
Add hours to worlddrivenbot's post-review message

### LINKED ISSUE #172: Post-review worlddrivenbot message doesn't mention the hours until merge
I noticed that the post-review message from worlddrivenbot only mentions the days until merge not the hours. I think adding the hours is a good idea.

I believe this section is the one that need updating: https://github.com/TooAngel/worlddriven/blob/cc450b37884b35eb2a6928e29ca8e25b6444f024/src/routes/githubWebHook.py#L139-L143

The hours can be calculated the same way as it is above, i.e `math.floor(pr.days_to_merge.seconds / 3600)`, but we can also get rid of the math import by using `pr.days_to_merge.seconds // 3600` instead.

Any thoughts?

### PR DESCRIPTION
**Pull request by IanFindlay**
Resolve #172 and:

Update math.floor() to Python's in-built floor divide and, as this was
the only use of the module, remove the import.

Update 'days' and 'hours' in the messages to 'day(s)' and 'hour(s)' to
indicate that the plural isn't needed when the integer is one.

### CODE REVIEW

**Comment by IanFindlay** [comment]
Never thought to check whether a small change like this would break a test, should have run them regardless my apologies. 

Although it has brought up a potential issue: having tests so dependent on specific strings means you have to rewrite both if you change them. Instead of checking the call is exact, could we not capture the call and use AssertIn statements to ensure it has the key pieces of information instead?

I'm happy to update the test to match this pull, but perhaps rewriting the test to be more flexible is the better option here.

**Comment by TooAngel** [comment]
IMHO in most (close to all) cases when you touch code you have to touch the tests. Texts/Strings are the only things that are debatable.
So, I'm fine with just testing the variables instead of the full string.

**Comment by IanFindlay** [comment]
I think for now I will just update the test call.

On a related note, how would I run these tests locally? I've got the virtual environment set up and gunicorn command from the README seems to work fine, but I'm not sure what I need to do in terms of the GitHub OAuth stuff or the mongodb backend aspects. Any help is much appreciated.

**Comment by TooAngel** [comment]
You can see the commands which are executed on the CI server here: https://github.com/TooAngel/worlddriven/blob/master/.circleci/config.yml

For testing the python code you run `py.test`, but they currently rely on an existing mongodb server.
The mongodb server is expected at `localhost:27017`, you can change that by setting `MONGODB_URI` as environment variable.

The PR https://github.com/TooAngel/worlddriven/pull/165 will introduce a docker environment at least for running the application.

Does that help?

**Comment by IanFindlay** [comment]
Very helpful thank you.

**Comment by TooAngel** [comment]
I like the changes, but I really would like to have the docker-compose (https://github.com/TooAngel/worlddriven/pull/164) merged before and it was already delayed due to other merges in between and merge conflicts.
So I will approve it afterwards.

**Comment by IanFindlay** [comment]
Sounds good to me.

**Review by TooAngel — APPROVED** [review]
Thx, I like

**Review by TooAngel — APPROVED** [review]

**Comment by IanFindlay** [comment]
@TooAngel I was wondering if there was a better way I could have done that merge? I just pressed the button to merge here but that obviously added another commit and so more time. 

On my local branch, I've rebased so that the first commit - "Add hours to..." squashes both my update to the test and the merge. i.e my branch history goes:

Add hours...

Merge pull request 158 from LeonKie/master

Would this be the correct way to approach this in the future? And if I were to push this now would the merge now happen as there will only be one commit now and the time had otherwise passed?

Thanks for any information on this, still trying to get my head around the worlddriven system.

**Comment by TooAngel** [comment]
Ok, this is a bit tricky to answer, but I will try:

So I like to have some kind of linear git history and due to time to merge increase (because of worlddriven) due to more commits, for smaller changes, it is obviously better to have just one commit. (if you have a bigger change, where multiple commits make sense, because of separated parts, of course, do multiple commits)

When one of my pull requests run into a merge conflict, I'm rebasing my changes to the origin/master `git rebase origin/master` (maybe with `-i` to squash if necessary) and force push the changes.
The time to merge is reset as soon as changes happen, to prevent that someone sneaks in nasty changes, like, waiting short before the time to merge and then force pushing some virus, ads, ...
TLDR if you push new commits, force push or do any other changes the time to merge is reset.
In this case, I think it would speed it up because 3 commits = 20 days, 1 commit = 10 days.
I hope this makes sense.

Now the tricky part:
In bigger open source projects it is usually not a good idea to change history (squashing or rebasing) as soon as your changes are public (pushed to a public repository), because someone else could have picked up your changes and is working on top of that. Changing history usually leads to a lot of merge conflict hassle for the other one.

There are ways around it, with e.g. fixup commits - So you know before that you will fixup and you can commit with `git commit --fixup HEAD~1`, but this needs a lot of foresight.

TLDR - I would suggest in this case to squash the commits and force push it. Just be aware that this is not always considered as the best practice.

**Comment by IanFindlay** [comment]
Thanks for the information. I'm going to force push my local branch, which I have squashed as described, and hopefully that will make this PR 1 commit as opposed to 3. I apologise in advance if I have done something wrong and this push turns out to be a disaster.

**Comment by TooAngel** [comment]
No worries, I'm happy that you like the worlddriven idea and you are willing to contribute. I'm happy to help.

The Pull Request seems a bit of and the `README.md` has a conflict.
When rebasing and squashing make sure you updated the index properly `git fetch --all --prune` (having the `tooangel/worlddriven` as a remote e.g. `worlddriven`) and rebase on top of that `git rebase worlddriven/master` (with `-i` in case you want to squash)

**Comment by IanFindlay** [comment]
Should be good now (I hope) and thank you again for all your help. I see the irony in being the owner of a project about automating things so projects don't need active maintainers and having to provide so much help to someone contributing to it.

**Review by TooAngel — APPROVED** [review]

**Comment by TooAngel** [comment]
Well, actually I think projects need to have active maintainers, someone needs to drive the project forward and keep the overview somehow. I think that maintainers are not a static list, it should be dynamic and that needs to be defined somehow.

I like to explain things (not necessary because of helping, more because I'm learning so much), also the interaction with worlddriven by different people in the last days (because of hacktoberfest) showed me different angles I have to think about (and certain things where I was just too lazy).

TLDR: Keep asking and pointing out, please :-)

**Comment by TooAngel** [comment]
This pull request was merged by [worlddriven](https://www.worlddriven.org).

**Pull request merged**
